server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -226,25 +230,25 @@ public SearchPhaseController.ReducedQueryPhase reduce() throws Exception {
             );
         }
         return reducePhase;
+
     }
 
     private static final Comparator<QuerySearchResult> RESULT_COMPARATOR = Comparator.comparingInt(QuerySearchResult::getShardIndex);
 
     private MergeResult partialReduce(
-        QuerySearchResult[] toConsume,
```

Comment:
why go from array to ArrayList here? How does this relate to nulling out the merge result?